### PR TITLE
Field Validation

### PR DESCRIPTION
**Pull request by chanchana**
### Description
- Add model's field validation
    - required, regex, unique
- Add rules to campaign, report and user model
- Add error alert on frontend, refactor error handling logic
    - Generally, frontend code's error handling is more organized

### Screenshots
Screen Shot 2021-07-31 at 8 46 08 PM

### CODE REVIEW

File: core/model.py [hunk 1]
```diff
@@ -120,7 +121,7 @@ def is_saved(self) -> bool:
         '''
         Retrun True if this record is saved in database.
         '''
-        return hasattr(self, 'id') and self.id
+        return bool(hasattr(self, 'id') and self.id)
```
Comment (justinasr) [inline core/model.py:124]: I think `bool` is redundant here? `and` should always return a boolean

**Review by justinasr — COMMENTED** [review]
One small comment

File: core/model.py (hunk 1 above)
Comment (chanchana) [reply core/model.py:124]: Yep. My first thought is the same as yours, until I debug and found that it return an actual id as string if the record is saved.

Comment (justinasr) [reply core/model.py:124]: Ah, yes, indeed, the thing where you can use `or` to return the truthy variable. `and` seems to have the same thing.

**Pull request merged**